packages/bierzo-wallet/src/communication/requestgenerators.ts

```diff
@@ -99,3 +114,21 @@ export const generateRegisterUsernameTxRequest = async (
     },
   };
 };
+
+export const generateUpdateUsernameTxRequest = async (
+  creator: Identity,
+  username: string,
+  targets: readonly ChainAddressPair[],
+): Promise<JsonRpcRequest> => {
+  const transactionWithFee = await generateUpdateUsernameTxWithFee(creator, username, targets);
+
+  return {
+    jsonrpc: "2.0",
+    id: makeJsonRpcId(),
+    method: "signAndPost",
+    params: {
+      reason: TransactionEncoder.toJson("I would like you to sign this request"),
+      transaction: TransactionEncoder.toJson(transactionWithFee),
```

Comment:
FYI: once IOV Core 2.x is used, the "signAndPost" RPC request needs a 3rd argument ("signer"). This cannot be checked by the type system. I'll update https://github.com/iov-one/ponferrada/pull/939 for this new code now.